lib/tasks/hybrid_cloud.rake

```diff
@@ -1,64 +1,122 @@
+  # The response will be logged, and any errors will be caught and logged as well.
+  # The task will exit with an error message if the organization does not have a manifest imported or if the token is not entered.
+  # The task will also log a warning if the custom URL is not set and the default one is used.
   task hybridcloud_register: [:environment] do
     include ::ForemanRhCloud::CertAuth
     include ::InsightsCloud::CandlepinCache
 
-    def logger
-      @logger ||= Logger.new(STDOUT)
+    # Helper method to get the registrations URL, with a warning for default usage
+    def registrations_url(custom_url)
+      if custom_url.empty?
+        logger.warn("Custom url is not set, using the default one: #{ForemanRhCloud.base_url}")
+        URI.join(ForemanRhCloud.base_url, '/api/identity/certificate/registrations').to_s
+      else
+        URI.join(custom_url, '/api/identity/certificate/registrations').to_s
+      end
     end
 
-    def registrations_url
-      logger.warn("Custom url is not set, using the default one: #{ForemanRhCloud.base_url}") if ENV['SATELLITE_RH_CLOUD_URL'].empty?
-      ForemanRhCloud.base_url + '/api/identity/certificate/registrations'
+    # --- Input Collection ---
+    puts "Paste in your organization ID, this can be retrieved with the command: hammer organization list"
+    loop do
+      input = STDIN.gets.chomp
+      if input.match?(/^\d+$/) # Checks if input consists only of digits
+        @user_org_id = input.to_i
+        break
+      else
+        puts "Invalid input. Please enter a numeric organization ID."
+      end
     end
 
-    if ENV['org_id'].nil?
-      logger.error('ERROR: org_id needs to be specified.')
-      exit(1)
-    end
+    puts "\n" + "-" * 50 + "\n\n"
+    puts 'Paste in your Custom Insights URL. If nothing is entered, the default will be used.'
+    insights_user_input = STDIN.gets.chomp
 
-    @organization = Organization.find_by(id: ENV['org_id'].to_i) # saw this coming in as a string, so making sure it gets passed as an integer.
-    @uid = cp_owner_id(@organization)
-    @hostname = ForemanRhCloud.foreman_host_name
-    logger.error('Organization provided does not have a manifest imported.') + exit(1) if @uid.nil?
+    # --- Data Preparation ---
 
-    puts 'Paste your token, output will be hidden.'
-    @token = STDIN.noecho(&:gets).chomp
-    logger.error('Token was not entered.') + exit(1) if @token.empty?
+    organization = Organization.find_by(id: @user_org_id)
 
-    def headers
-      {
-        Authorization: "Bearer #{@token}",
-      }
+    if organization.nil?
+      logger.error("Organization with ID '#{@user_org_id}' not found.")
+      exit(1)
     end
 
-    def payload
-      {
-        "uid": @uid,
-        "display_name": "#{@hostname}+#{@organization.label}",
-      }
+    uid = cp_owner_id(organization)
+    if uid.nil?
+      logger.error('Organization provided does not have a manifest imported.')
```

Comment:
Could we add the organization name in this message? Something like:

```ruby
logger.error("Organization '#{organization}' does not have a manifest imported.")
```

Reply:
Fixed in our sync up